tests/test_dict_like_matrix_wrapper.py

```diff
@@ -6,54 +6,52 @@
 import sys
 import unittest
 
+from numpy import float64
+
 from prog_models.utils.containers import DictLikeMatrixWrapper
 from prog_models import ProgModelTypeError
 
 
 class TestDictLikeMatrixWrapper(unittest.TestCase):
     def setUp(self):
-        # set stdout (so it wont print)
+        # set stdout (so it won't print)
         sys.stdout = StringIO()
 
     def tearDown(self):
         sys.stdout = sys.__stdout__
-    
+
     def _checks(self, c1):
         self.assertListEqual(c1.keys(), ['a', 'b'])
-        self.assertListEqual(list(c1.values()), [1, 2])
-        self.assertListEqual(list(c1.items()), [('a', 1), ('b', 2)])
+        print(list(c1.values()), [1., 2.])
+        self.assertListEqual(list(c1.values()), [1., 2.])
+        self.assertListEqual(list(c1.items()), [('a', 1.), ('b', 2.)])
         self.assertEqual(c1['a'], 1)
         self.assertEqual(c1['b'], 2)
-        self.assertTrue((c1.matrix == np.array([[1], [2]])).all())
-
+        self.assertTrue(np.array_equal(c1.matrix, np.array([[1.], [2.]])))
         # Membership
         self.assertIn('a', c1)
         self.assertIn('b', c1)
         self.assertNotIn('c', c1)
         self.assertEqual(len(c1), 2)
-
         # Setting by dict
-        c1['a'] = -1
-        self.assertTrue((c1.matrix == np.array([[-1], [2]])).all())
-        self.assertEqual(c1['a'], -1)
-        
+        c1['a'] = -1.0
+        self.assertTrue((c1.matrix == np.array([[-1.], [2.]])).all())
+        self.assertEqual(c1['a'], -1.0)
         # Setting by matrix
-        c1.matrix[1][0] = -2
-        self.assertTrue((c1.matrix == np.array([[-1], [-2]])).all())
-        self.assertEqual(c1['b'], -2)
-
+        c1.matrix[1][0] = -2.0
+        c1.update({'b': -2.0})
+        self.assertTrue(np.all(c1.matrix == np.array([[-1.], [-2.]])))
+        self.assertEqual(c1['b'], -2.0)
```

Comment:
```suggestion
        c1.matrix[1][0] = -2.0
        self.assertTrue(np.all(c1.matrix == np.array([[-1.], [-2.]])))
        self.assertEqual(c1['b'], -2.0)
        
        # Test update function
        c1.update({'b': -2.0})
        self.assertTrue(np.all(c1.matrix == np.array([[-1.], [-2.]])))
        self.assertEqual(c1['b'], -2.0)
```

The way it's currently done doesn't test .matrix setting. Instead it tests that either matrix or update works.